plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageService.java

```diff
@@ -249,6 +272,11 @@ private BlobServiceClientBuilder applyLocationMode(final BlobServiceClientBuilde
 
     private static BlobServiceClientBuilder createClientBuilder(AzureStorageSettings settings) throws InvalidKeyException,
         URISyntaxException {
+        if (settings.usesManagedIdentityCredential()) {
```

Comment:
Please add validation for the simple string:

```suggestion
    /** Azure token credentials such as Managed Identity */
    public static final AffixSetting<String> TOKEN_CREDENTIAL_TYPE_SETTING = Setting.affixKeySetting(
        AZURE_CLIENT_PREFIX_KEY,
        "token_credential_type",
        key -> Setting.simpleString(key,
                value -> {
                    if (value != null && !value.equals("managed")) {
                        throw new IllegalArgumentException("The token credential type " + key 
                            + " is unsupported, please use one of the following values [managed]");
                    }
                },
                Property.NodeScope),
        () -> ACCOUNT_SETTING
    );

```